tests/nested/manual/core20-factory-reset-install-device-hook/task.yaml

```diff
@@ -0,0 +1,181 @@
+    unsquashfs -d pc-gadget pc.snap
+
+    echo "Sign the shim binary"
+    tests.nested secboot-sign gadget pc-gadget "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+
+    echo "Add the install-device and prepare-device hooks"
+    mkdir -p pc-gadget/meta/hooks
+    cp install-device pc-gadget/meta/hooks/install-device
+    cp prepare-device pc-gadget/meta/hooks/prepare-device
+
+    echo "Add the extra hooks definition to the snap.yaml"
+    # convert our yaml files to json (yaml is easier to write and maintain in VCS)
+    yaml2json < pc-gadget/meta/snap.yaml | tee pc-gadget/meta/snap.json > /dev/null
+    yaml2json < snap-yaml-extras.yaml | tee snap-yaml-extras.json > /dev/null
+    # slurp our two json files together into one json document, then convert 
+    # back to yaml and write out to the snap.yaml in the unpacked gadget snap
+    jq -s '.[0] * .[1]' <(cat snap-yaml-extras.json) <(cat pc-gadget/meta/snap.json) | json2yaml | tee pc-gadget/meta/snap.yaml
+
+    # delay all refreshes for a week from now, as otherwise refreshes for our 
+    # snaps (which are asserted by the testrootorg authority-id) may happen, which
+    # will break things because the signing keys won't match, etc. and 
+    # specifically snap-bootstrap in the kernel snap from the store won't trust
+    # the seed keys to unlock the encrypted data partition in the initramfs
+    sed defaults.yaml -e "s/HOLD-TIME/$(date --date="next week" +%Y-%m-%dT%H:%M:%S%:z)/" >> \
+      pc-gadget/meta/gadget.yaml
+
+    snap pack pc-gadget/ "$(tests.nested get extra-snaps-path)"
+
+    # start fake device svc
+    systemd-run --collect --unit fakedevicesvc fakedevicesvc localhost:11029
+
+    # include the extra snap declaration bits we need to auto-connect the 
+    # system-files plugs for the install-device hook
+    # XXX: this is a bit of a hack, we want the snap to be able to use the
+    # system-files interface from the install-device hook, so here we just allow
+    # any system-files interface plug to be used with the snap, in reality this
+    # rule would be much more specific
+
+    NESTED_FAKESTORE_SNAP_DECL_PC_GADGET="pc-snap-decl-extras.json"
+    export NESTED_FAKESTORE_SNAP_DECL_PC_GADGET
+    tests.nested build-image core
+    unset NESTED_FAKESTORE_SNAP_DECL_PC_GADGET
+
+    tests.nested create-vm core
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    # stop fake device svc
+    systemctl stop fakedevicesvc
+
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+    teardown_fake_store "$NESTED_FAKESTORE_BLOB_DIR"
+
+execute: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    # wait for the initialize device task to be done
+    retry -n 200 --wait 1 sh -c "tests.nested exec snap changes | MATCH 'Done.*Initialize device'"
```

Comment:
(nitpick) maybe make the wait 2-5s to avoid too much spam on the terminal while this waits.

Reply:
No worries! Increased this to 3s - thanks!

Reply:
is there an original test(s) with the same problem? we should tweak them all I suppose

Reply:
Yea exactly, I changed the original test as well